CHANGELOG.md

```diff
@@ -1,16 +1,76 @@
+-   The methods `intercept` and `observe` are no longer exposed on observable arrays, maps and boxed observables. Import them as utility from mobx instead: `import { observe, intercept } from "mobx"`, and pass the collection as first argument: `observer(collection, callback)`. Note that we still recommend to avoid these API's.
+-   `observableMap.toPOJO()`, `observableMap.toJS()` have been dropped. use `new Map(observableMap)` instead if you want to convert an observable map to a plain Map shallowly.
+-   `observableMap.toJSON()` now returns an entries array rather than a new Map, to better support serialization.
+-   `observableSet.toJS()` has been dropped. Use `new Set(observableSet)` instead if you want to convert an observable Set to a plain Set shallowly.
+-   `observableMap.toJSON()` now returns an array rather than a new Set, to better support serialization.
+-   Sorting or reversing an observableArray in a derivation (without slicing first) will now throw rather than warn. In contrast, it is now allowed to sort or reverse observable arrays in-place, as long as it happens in an action.
+-   `isArrayLike` is no longer exposed as utility. Use `Array.isArray(x) || isObservableArray(x)` instead.
+
+### Obscure things that don't work anymore, but that probably won't affect you
+
+-   It is no longer possible to re-decorate a field (through either `@observable` or `makeObservable`) that is already declared in a super class.
+-   `runInAction` no longer supports passing a name as first argument. Name the original function or use `action(name, fn)()` if you cared about the debug name.
+-   `computed(getterFn, setterFn)` no longer accepts a setter function as a second argument. Use the `set` option instead: `computed(getterFn, { set: setterFn })`.
+-   In observable arrays, for `findIndex` / `find` method, the `offset` argument (the third one) is no longer supported, to be consistent with ES arrays.
+-   The option `computedConfigurable` of `configure` is no longer supported as it is now the default.
+-   `observableArray.toJS()` has been removed, use `observableArray.slice()` instead, which does the same.
+-   Killed support for the `IGNORE_MOBX_MINIFY_WARNING` environment flag.
+-   `_allowStateChangesInComputation(fn)` is no longer needed, use `runInAction(fn)` instead.
+-   In `computed`, the when `predicate` (first arg), and `reaction` predicate (first arg) it is now forbidden to directly change state. State changes should be done in their effect functions, or otherwise at least wrapped in `runInAction` (only the state change, not the observables you want to track!). Note that this is still an anti pattern.
```

Comment:
```suggestion
-   In `computed`, the `when` predicate (first arg), and `reaction` predicate (first arg) it is now forbidden to directly change state. State changes should be done in their effect functions, or otherwise at least wrapped in `runInAction` (only the state change, not the observables you want to track!). Note that this is still an anti-pattern.
```